cypress/pages/BounceRulesContainer/index.js

```diff
@@ -82,42 +79,82 @@ class BounceRulesPage extends Page {
     return cy.get(Selectors.confirmationSubmit);
   }
 
-  get testRule() {
-    return cy.get("[data-rule='cypressTest']");
+  get createdDeleteRuleAPI() {
+    return cy.get('[data-cypress="cypressDeleteTest"]');
   }
 
-  fillForm(
-    priority,
-    bounceAction,
-    responseCode,
-    description,
-    enhancedCode,
-    regex
-  ) {
-    if (priority) {
-      this.priority.type(priority);
-    }
-
-    if (bounceAction) {
-      this.bounceAction.type(bounceAction);
-    }
-
-    if (responseCode) {
-      this.responseCode.type(responseCode);
-    }
-
-    if (description) {
-      this.description.type(description);
-    }
-    if (enhancedCode) {
-      this.enhancedCode.type(enhancedCode);
-    }
+  get createdCreateRuleAPI() {
+    return cy.get('[data-cypress="cypressCreateTest"]');
+  }
 
-    if (regex) {
-      this.regex.type(regex);
-    }
+  open() {
+    super.open("/bounce_rules");
+  }
 
-    return this.submitButton.click();
+  createdRuleButton(id) {
+    return cy.get(`[data-rule="${id}"]`);
+  }
+
+  deleteBounceRuleAPI(testRule) {
+    return new Promise(resolve => {
+      cy.task("getRules").then(res => {
+        if (!res.errno) {
+          for (let i = 0; i < res.length; i++) {
+            if (_.isEqual(testRule, _.omit(res[i], "id"))) {
+              cy.task("deleteRule", res[i].id);
+              cy.log("DELETE SUCCESS");
+            }
+          }
```

Comment:
It's also technically a success if none of the bounce rules match the one you were deleting since it never existed so you would log there in that case